lib/auth/webauthncli/fido2.go

```diff
@@ -526,16 +526,28 @@ func runOnFIDO2Devices(
 	prompt runPrompt, passwordless bool,
 	filter deviceFilterFunc,
 	deviceCallback deviceCallbackFunc) error {
-	devices, err := findSuitableDevicesOrTimeout(ctx, filter)
+	// Do we have readily available devices?
+	prompted := false
+	devices, err := findSuitableDevices(filter, make(map[string]struct{}))
+	if err != nil {
+		// No readily available devices means we need to prompt, otherwise the
+		// user gets no feedback whatsoever.
+		prompt.PromptTouch()
+		prompted = true
+
+		devices, err = findSuitableDevicesOrTimeout(ctx, filter)
+	}
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
 	var dev deviceWithInfo
-	if shouldDoEagerPINPrompt(passwordless, devices) {
+	if !prompted && shouldDoEagerPINPrompt(passwordless, devices) {
```

Comment:
I'm asking UX about changing the prompt from "Tap your security key" to "Insert and tap your security key". If we do, then asking for the PIN right away wouldn't look so weird, so I could revert this line.

Ie, it becomes something like:

```
$ tsh login
> Insert and tap your security key
*user plugs PIN device*
> Enter your security key PIN:
(...)
```